pkg/kubelet/cm/deviceplugin/manager.go

```diff
@@ -610,58 +610,95 @@ func (m *ManagerImpl) allocateContainerResources(pod *v1.Pod, container *v1.Cont
 		if allocDevices == nil || len(allocDevices) <= 0 {
 			continue
 		}
-		startRPCTime := time.Now()
-		// devicePluginManager.Allocate involves RPC calls to device plugin, which
-		// could be heavy-weight. Therefore we want to perform this operation outside
-		// mutex lock. Note if Allocate call fails, we may leave container resources
-		// partially allocated for the failed container. We rely on updateAllocatedDevices()
-		// to garbage collect these resources later. Another side effect is that if
-		// we have X resource A and Y resource B in total, and two containers, container1
-		// and container2 both require X resource A and Y resource B. Both allocation
-		// requests may fail if we serve them in mixed order.
-		// TODO: may revisit this part later if we see inefficient resource allocation
-		// in real use as the result of this. Should also consider to parallize device
-		// plugin Allocate grpc calls if it becomes common that a container may require
-		// resources from multiple device plugins.
+		// Update internal cached podDevices state.
 		m.mutex.Lock()
-		e, ok := m.endpoints[resource]
+		m.podDevices.insert(podUID, contName, resource, allocDevices, nil)
```

Comment:
Given that we don't need to cache Allocate response, we should remove the last parameter.

Reply:
Do you mean not needed because we would make Allocate each time?

Reply:
@ScorpioCPH Please not that last parameter, `resp`, is nil here and in `GetDeviceRunContainerOptions`, `resp` will be updated. So i would say it will be "updated".